projects/igniteui-angular/src/lib/grids/cell.component.ts

```diff
@@ -255,7 +255,7 @@ export class IgxGridCellComponent implements OnInit, OnChanges, OnDestroy, CellT
      * @memberof IgxGridCellComponent
      */
     public get context(): IgxCellTemplateContext {
-        const getCellType = () => this.getCellType(true);
+        const getCellType = () => this.editMode ? this.grid.crudService.cell : this.getCellType(true);
```

Comment:
This looked initially to me like treating the symptom, rather than fixing the core issue. Debugging into `this.getCellType` I stumbled upon code with the same meaning:

![image](https://github.com/IgniteUI/igniteui-angular/assets/14248932/f7bdd488-c088-4fef-b739-6742ff476e05)

It feels unnatural to introduce the second code here, when it is expected t get the correct info by using `this.getCellType`. So let;s change the fix to have `this.getCellType` return correct result.

Reply:
Since the IgxGridCell constructor accept column field name or a column component 
![image](https://github.com/IgniteUI/igniteui-angular/assets/34240583/9ec411cc-63e8-4796-9c94-416ab06c5c8d)
 I can suggest just to pass the correct column when the grid cell type is created in the getCellType method... because we do not need to search for the column in the column list when the cell holds that reference already

Reply:
> Since the IgxGridCell constructor accept column field name or a column component ![image](https://user-images.githubusercontent.com/34240583/263712176-9ec411cc-63e8-4796-9c94-416ab06c5c8d.png) I can suggest just to pass the correct column when the grid cell type is created in the getCellType method... because we do not need to search for the column in the column list when the cell holds that reference already

That's true and will solve it, but will still hide that the column getter may return incorrect info:
![image](https://github.com/IgniteUI/igniteui-angular/assets/14248932/7c7e9c96-13d4-4df0-b34a-ee6cebf14bab)

Reply:
So I think one does not interfere with the other. Hrisi, I think you can go this route:
1) Fix the column getter to return correct info
2) When creating an instance of IgxGridCell, if possible pass the column object instead of the column field
3) Find other places in the code where column is searched by field name using find and see if it leads to errors